### PR TITLE
Support caching album items

### PR DESCRIPTION
**Pull request by stulees**
## Description
For the `albumItems` endpoint, service now saves data to cache and responds with cached data if the cache expiry has not exceeded. Further details as follows:

- Cache expiry is `60` minutes. The `cache-control` header specifies a `max-age` value for component (rise-folder) to adhere to. 
- A `lastUpdated` object is added to `status` and is used for saving a timestamp per album id and thereby informing if cache for that album id items is current for the request and aids in the calculation of `max-age` value for `cache-control` header. 
- Set up support for future `count` param for `albumItems` endpoint (last added feature). Temporarily forcing a `count` value to be the API maximum album items returned (100).
- If a concurrent request is already in progress for `albumItems`, cached data is returned for that album if available. Since updated data is in progress, the `max-age` value for `cache-control` header is set to a low expiry of `30` seconds to help component request again sooner than later. 
- If a request is for all photos, as in no `albumId` is provided in request to `albumItems` endpoint, the data is still cached under the same conditions. A special value `all-photos` is used for saving to cache to identify and differentiate in cache from specific album id results stored. 

## Motivation and Context
Photos Pitch

## How Has This Been Tested?
Directly in browser and in apps staging. Most tests were done with a forced `3` minute expiry prior to finalizing `60` minutes. 

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed.
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by stulees** [comment]
@olegrise Please review, thanks

File: src/config.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
   defaultPort: 80,
   redisCacheHostname: "gs-redis-master",
   redisOtpHostname: "otp-redis-master",
   concurrentRequestTimeoutInMillis: 2 * MINUTES,
+  photosApiAlbumsMax: 50,
+  photosApiAlbumItemsMax: 100,
+  albumItemsCacheExpirationInMillis: 1 * HOURS,
+  /* client will be asked to retry after this time to see if processing flag has been cleared */
+  retryRequestInSeconds: 30,
+  allPhotosId: "all-photos",
```
Comment (olegrise) [inline src/config.js:14]: @stulees what do you think about making this cache key more unique like "-=#@ all-photos @#=-" so there is a less chance that user may actually have an album with name like that?

File: src/redis-cache/api.js [hunk 2]
```diff
@@ -16,7 +17,30 @@ const saveStatus = (account, endpoint, status) => {
   return redis.setString(statusKeyFor(account, endpoint), value);
 };
 
+const getAlbumItemsFor = (account, albumId, count) => {
+  const key = albumItemsKeyFor(account, albumId);
+
+  return redis.getListRange(key, 0, count - 1).then((items) => {
```
Comment (olegrise) [inline src/redis-cache/api.js:23]: Interesting.

**Review by olegrise — APPROVED** [review]
@stulees just one minor comment based on not very thorough review.

File: src/config.js (hunk 1 above)
Comment (stulees) [reply src/config.js:14]: @olegrise I'm good with that. Can I leave it with you to do in a future PR?

**Comment by olegrise** [comment]
@stulees If I understand the logic correctly, there are 2 entities that are being cached - `status` and `albumItems`. The `status` is per `account`+`endpoint`. The `albumItems` is per `account`+`albumId`. Can you clarify why `lastUpdated` value is saved as "array" in `status` entity and not as a plain value in `albumItems` entity? Thanks.

**Comment by olegrise** [comment]
@stulees we discussed that `revokeAccess` should clear cached data. Was that not added to this PR? Thanks.

File: src/config.js (hunk 1 above)
Comment (olegrise) [reply src/config.js:14]: @stulees I was wrong. I was thinking about album names, but we use album ID(s). No need to change anything. Thanks.

**Pull request merged**